core/arch/arm/include/kernel/thread.h

```diff
@@ -247,6 +247,7 @@ struct thread_specific_data {
 #ifdef CFG_CORE_DEBUG_CHECK_STACKS
 	bool stackcheck_recursion;
 #endif
+	unsigned int recursion_depth;
```

Comment:
`syscall_recursion` to better reflect the `CFG_` name?

Reply:
Makes sense, I'll fix.